sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -206,6 +206,9 @@ case class AlterTableAddColumnsCommand(
       reorderedSchema.map(_.name), "in the table definition of " + table.identifier,
       conf.caseSensitiveAnalysis)
 
+    val newDataSchema = StructType(catalogTable.dataSchema ++ columns)
+    DDLUtils.checkFieldNames(catalogTable.copy(schema = newDataSchema))
```

Comment:
For this command, it's not easy to get `CatalogTable` at `DataSourceStrategy`.

Reply:
This should be moved to `verifyAlterTableAddColumn`

Reply:
```Scala
    val reorderedSchema = catalogTable.dataSchema ++ columns ++ catalogTable.partitionSchema
    val newSchema = catalogTable.schema.copy(fields = reorderedSchema.toArray)

    SchemaUtils.checkColumnNameDuplication(
      reorderedSchema.map(_.name), "in the table definition of " + table.identifier,
      conf.caseSensitiveAnalysis)
    DDLUtils.checkFieldNames(catalogTable.copy(schema = newSchema))

    catalog.alterTableSchema(table, newSchema)
```

Reply:
Ur, actually. Excluding partition columns was intentional.
Maybe, I used a misleading PR title and description here.
So far, I checked `dataSchema` only. I think partition columns are okay because they are not a part of Parquet/ORC file schema.

Reply:
Is it okay to use the following?
```scala
    val reorderedSchema = catalogTable.dataSchema ++ columns ++ catalogTable.partitionSchema
    val newDataSchema = StructType(catalogTable.dataSchema ++ columns)

    SchemaUtils.checkColumnNameDuplication(
      reorderedSchema.map(_.name), "in the table definition of " + table.identifier,
      conf.caseSensitiveAnalysis)
    DDLUtils.checkFieldNames(catalogTable.copy(schema = newDataSchema))

    catalog.alterTableSchema(
      table, catalogTable.schema.copy(fields = reorderedSchema.toArray))
```

Reply:
Sorry, I found that your code is better. I'll updated it like yours.